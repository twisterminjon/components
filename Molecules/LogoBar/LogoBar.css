.logobar-wrap {
  position: relative;
  display: flex;
  align-items: center;

  height: var(--searchbar-height);
  width: 100%;
  z-index: var(--z-logobar);

  background-color: var(--white);
}
