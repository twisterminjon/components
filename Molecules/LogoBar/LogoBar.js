import React, { Component } from 'react'
import PropTypes from 'prop-types'

import EnterpriseLogo from '../../Atoms/EnterpriseLogo/EnterpriseLogo'

import './LogoBar.css'

export default class LogoBar extends Component {
  static propTypes = {
    /** Path to image file for the enterprise logo */
    enterpriseLogo: PropTypes.string.isRequired,
  }
  static defaultProps = {
    enterpriseLogo: '',
  }

  render() {
    const { enterpriseLogo } = this.props

    return (
      <div className="logobar-wrap">
        <EnterpriseLogo imgUrl={enterpriseLogo} />
      </div>
    )
  }
}
