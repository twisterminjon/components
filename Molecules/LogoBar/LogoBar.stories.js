import React from 'react'

import { storiesOf } from '@storybook/react'
import { withKnobs, boolean } from '@storybook/addon-knobs'

import LogoBar from './LogoBar'

const stories = storiesOf('Provider/v1/Molecules', module)
stories.addDecorator(withKnobs)

stories.add('LogoBar', () => {
  const picLabel = 'Show image'
  const picDefaultValue = true
  const picValue = boolean(picLabel, picDefaultValue)

  const picSource = picValue ? 'https://www.fillmurray.com/600/200' : ''

  return (
    <div
      style={{
        position: 'fixed',
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
        backgroundColor: 'cornsilk',
      }}>
      <LogoBar enterpriseLogo={picSource} />
    </div>
  )
})
