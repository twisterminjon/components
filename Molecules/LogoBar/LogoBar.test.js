import React from 'react'
import LogoBar from './LogoBar'

describe('LogoBar', () => {
  const mockFun = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<LogoBar onClick={mockFun} />)

    expect(wrapper).toMatchSnapshot()
  })
})
