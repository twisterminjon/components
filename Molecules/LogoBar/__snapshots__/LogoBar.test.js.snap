// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoBar matches the snapshot 1`] = `
<div
  className="logobar-wrap"
>
  <EnterpriseLogo
    imgUrl=""
  />
</div>
`;
